lib/thredded/per_notifier_pref.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+module Thredded
+  # in this class we dynamically set up an accessor for all the Thredded.notifiers (by their key)
+  # as we don't know in advance what the keys are,
+  # we try to not pollute the namespace
+  # nonetheless some keys would be bad, so we sanity check those on setup
+  class PerNotifierPref
```

Comment:
this is substantially done now. Just doing some tidying and validation for notifiers . FYI: We are close to being able to specify per notifier defaults.